src/core/model.js

```diff
@@ -136,6 +140,7 @@ export default class Model extends Object3D {
   /* eslint-enable complexity */
 
   destroy() {
+    removeModel(this.id);
```

Comment:
As discussed this method is rarely (if ever) called today. But can't really see other ways of doing it. 

Maybe this is a motivation to improve deck.gl in this regard.